teps/0089-nonfalsifiable-provenance-support.md

```diff
@@ -59,26 +63,41 @@ There are a couple issues with this:
+A SPIRE deployment relies on two key components, the SPIRE server, and its associated SPIRE agents.
+- The SPIRE server serves as a central management system for SPIRE, responsible for interfacing with any key material, authorities, and databases. It is also responsible for attesting the agents that are part of its trust domain.
+- The SPIRE server acts as local endpoints for each compute node that is part of the SPIRE deployment.
 
-SPIRE can be used to request certificates (SVIDs) for a given k8s workload and can use these certificates to create signatures for the data we need to verify, i.e. an entire TaskRun yaml and results. For our use case, both the Tekton entrypointer image (running in Pods) and the Tekton Controller will interact with SPIRE.
+Pods running in a cluster can interact with SPIRE through SPIFFE api via the local SPIRE agent socket.
+
+Getting pod identity
+- A pod can request for an identity by talking to the local agent socket. The SPIRE agent attests the identity of the pod,
+and request for an identity for the pod to the SPIRE server.
+- The workload then receives one or more SPIFFE Verifiable Identity Document, or [SVID](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/#spiffe-verifiable-identity-document-svid), an X509 certificate together with the associated private key for the pod's identity. The x509 certificate is signed by the SPIRE server's certificate authority.
+
+Verifying pod identity
+- A pod can request for an trust bundle of a SPIRE server, which will include the public key to validate the pod's certificate.
+- This trust bundle can be used to verify the SVID of a pod
+
+Note: Alternatively, using the JWT SPIFFE endpoint, one can request a JWT with a specified audience field. Although possible, it may not be idiomatic to use the audience for payload signing purposes. However, there is an open [issue](https://github.com/spiffe/spire/issues/1848) discussing adding arbitrary claims as well.
+
+Identity registration is an important aspect of getting SPIRE identities. This defines the subject name of identities that are minted to pods, as well as the attestation requirements of the workloads. This can be done specific to usecase, or there is a kubernetes workload registrar that creates identity based on the fully qualified canonical pod name.
+
+### Using SPIRE for provenance
+
+SPIRE can be used for provenance using the private key provided from the workload's SVID and signing a payload. The signature will be verifiable by the x509 certificate of the SVID together with the trust bundle of the SPIRE deployment. The verification would ensure that the payload's provenance comes from the Tekton entrypointer image (running in Pods) or the Tekton Controller.
```

Comment:
nit -> "can be used for signing provenance" ? (i assume we're not saying SPIRE would be creating the provenance)

Reply:
Yup changed!